### PR TITLE
add `soft_delete_retention_days` support for `azurerm_key_vault`

### LINKED ISSUE #6665: Support for [configuring retention period on azure key vault]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Please add the argument for configuring retention days on key vault when soft delete is enabled.

### New or Affected Resource(s)

* azurerm_key_vault 

### Potential Terraform Configuration

resource "azurerm_resource_group" "example" {
  name     = "resourceGroup1"
  location = "West US"
}

resource "azurerm_key_vault" "example" {
  name                        = "testvault"
  location                    = azurerm_resource_group.example.location
  resource_group_name         = azurerm_resource_group.example.name
  enabled_for_disk_encryption = true
  tenant_id                   = data.azurerm_client_config.current.tenant_id
  soft_delete_enabled         = true
  purge_protection_enabled    = false
  **retention_days                        = "7"**

  sku_name = "standard"

  access_policy {
    tenant_id = data.azurerm_client_config.current.tenant_id
    object_id = data.azurerm_client_config.current.object_id

    key_permissions = [
      "get",
    ]

    secret_permissions = [
      "get",
    ]

    storage_permissions = [
      "get",
    ]
  }

### References

* https://docs.microsoft.com/en-us/cli/azure/keyvault?view=azure-cli-latest#az-keyvault-create

### LINKED ISSUE #7178: Support for softDeleteRetentionInDays on Key Vault
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently we have the ability to enable Soft Delete and Purge Protection on a Key Vault resource. The API (2019-09-01) from Microsoft exposes a softDeleteRetentionInDays that can take a value >= 7 and <= 90. Being able to set this would be very beneficial, as the default of 90 days can be quite long.

### New or Affected Resource(s)

azurerm_key_vault

### Potential Terraform Configuration

soft_delete_retention_days

```hcl
resource "azurerm_key_vault" "example" {
  name                        = "testvault"
  location                    = azurerm_resource_group.example.location
  resource_group_name         = azurerm_resource_group.example.name
  enabled_for_disk_encryption = true
  tenant_id                   = data.azurerm_client_config.current.tenant_id

  soft_delete_enabled         = true
  purge_protection_enabled    = true
  soft_delete_retention_days  = 7

  sku_name = "premium"

  access_policy {
    tenant_id = data.azurerm_client_config.current.tenant_id
    object_id = data.azurerm_client_config.current.object_id

    key_permissions = [
      "get",
    ]

    secret_permissions = [
      "get",
    ]

    storage_permissions = [
      "get",
    ]
  }

  network_acls {
    default_action = "Deny"
    bypass         = "AzureServices"
  }
}
```

### References

* https://docs.microsoft.com/en-us/azure/templates/microsoft.keyvault/vaults

### PR DESCRIPTION
**Pull request by njuCZ**
fix #6665 
fix #7178

this is a new field in api-version: 2019-09-01
once `soft_delete_retention_days` is set, it could not be updated

if `soft_delete_enabled` is true, and not set `soft_delete_retention_days`, the resp `softDeleteRetentionInDays`  is also nil, so no influence to existing test case.

### CODE REVIEW

File: website/docs/r/key_vault.html.markdown [hunk 1]
```diff
@@ -111,6 +112,10 @@ The following arguments are supported:
 
 ~> **Note:** Terraform will check when creating a Key Vault for a previous soft-deleted Key Vault and recover it if one exists. You can configure this behaviour using the `features` block within the `provider` block.  
 
+* `soft_delete_retention_days` - (Optional) The key Vault retention days when enable soft delete.
```
Comment (tombuildsstuff) [inline website/docs/r/key_vault.html.markdown:115]:
```suggestion
* `soft_delete_retention_days` - (Optional) The number of days that items should be retained for once soft-deleted.
```

File: website/docs/r/key_vault.html.markdown [hunk 2]
```diff
@@ -111,6 +112,10 @@ The following arguments are supported:
...
+
+!> **Note:** Once Soft Delete has been Enabled it's not possible to update it.
```
Comment (tombuildsstuff) [inline website/docs/r/key_vault.html.markdown:117]: can we make this:

```suggestion
~> **Note:** This field can only be set once Soft Delete is Enabled and cannot be updated.
```

**Comment by njuCZ** [comment]
@tombuildsstuff I have modified this PR, could you please have a look again?

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2020-08-27 at 10 29 22

**Pull request merged**